server/controllers/payroll/multiplePayroll/index.js

```diff
@@ -0,0 +1,1177 @@
+
+            if (taxesContributions.length) {
+              taxesContributions.forEach(taxContribution => {
+              // Set the result of IPR calculation
+                if (taxContribution.is_ipr) {
+                  taxContribution.result = iprValue;
+                }
+
+                // Calculation of the sum of taxes and membership fee borne by the employee
+                if (taxContribution.is_employee) { sumTaxContributionEmp += taxContribution.result; }
+
+                allRubrics.push([uid, taxContribution.rubric_payroll_id, taxContribution.result]);
+              });
+            }
+          } else if (taxesContributions.length) {
+            taxesContributions.forEach(taxContribution => {
+              // Calculation of the sum of taxes and membership fee borne by the employee
+              if (taxContribution.is_employee) {
+                sumTaxContributionEmp += taxContribution.result;
+              }
+
+              allRubrics.push([uid, taxContribution.rubric_payroll_id, taxContribution.result]);
+            });
+
+          }
+
+          const netSalary = grossSalary - sumTaxContributionEmp;
+
+          const paiementData = {
+            uuid : uid,
+            employee_uuid : db.bid(employee.uuid),
+            payroll_configuration_id : payrollConfigurationId,
+            currency_id : data.currency_id,
+            basic_salary : basicSalary,
+            base_taxable : baseTaxable,
+            daily_salary : dailySalary,
+            total_day : totalDayPeriod,
+            working_day : data.working_day,
+            gross_salary : grossSalary,
+            net_salary : netSalary,
+            amount_paid : 0,
+            status_id : 2,
+          };
+
+          const deletePaiementData = 'DELETE FROM paiement WHERE employee_uuid = ? AND payroll_configuration_id = ?';
+          const setPaiementData = 'INSERT INTO paiement SET ?';
+          const setRubricPaiementData = `INSERT INTO rubric_paiement (paiement_uuid, rubric_payroll_id, value) 
+            VALUES ?`;
+          const setHolidayPaiement = `INSERT INTO holiday_paiement 
+            (holiday_id, holiday_nbdays, holiday_percentage, paiement_uuid, label, value) VALUES ?`;
+          const setOffDayPaiement = `INSERT INTO offday_paiement 
+            (offday_id, offday_percentage, paiement_uuid, label, value) VALUES ?`;
+
+          transaction
+            .addQuery(deletePaiementData, [db.bid(employee.uuid), payrollConfigurationId])
+            .addQuery(setPaiementData, [paiementData]);
+
+          if (allRubrics.length) {
+            transaction.addQuery(setRubricPaiementData, [allRubrics]);
+          }
+
+          if (holidaysElements.length) {
+            transaction.addQuery(setHolidayPaiement, [holidaysElements]);
+          }
+
+          if (offDaysElements.length) {
+            transaction.addQuery(setOffDayPaiement, [offDaysElements]);
+          }
+
+          transaction.execute()
```

Comment:
You should `return transcation.execute()` here and move the `.then().catch()` code outside the promise chain.